anesthetic/samples.py

```diff
@@ -611,6 +611,38 @@ def dlogX(self, nsamples=None):
         else:
             return WeightedDataFrame(dlogX, self.index, weights=self.weights)
 
+    def importance_reweighting(self, logL_new, replace=True, add=False):
+        """Perform importance re-weighting on the log-likelihood.
+
+        Parameters
+        ----------
+        logL_new: np.array
+            New log-likelihood values. Should have the same shape as `logL`.
+
+        replace: bool
+            Replace the current `logL` with the new `logL_new`.
+
+        add: bool
+            Add the new `logL_new` to the current `logL`.
```

Comment:
I've included these two bool args to allow the user to decide whether to completely replace the `logL` or whether to add to it. 
Happy with that? 
Would there be a crisper way of implementing this that is easy to understand?

Reply:
What happens when they are both true or both false? I'm not sure but I'd probably favor other options:

- a string might be clearer, something like `importance_reweighting(..., action='add')`` or `importance_reweighting(..., action='replace')'
- two separate different methods with very clear names `loglike_replace(loglike)` and `loglike_add(loglike)`

Reply:
> What happens when they are both true or both false?

In that case I am raising an exception that let's the user know that one and only one of those two should be True.

Reply:
> * two separate different methods with very clear names loglike_replace(loglike) and loglike_add(loglike)

I think it would be nicer to keep this in one place.

> * just a single bool, since in the end it's a choice between two things

I went with the two args to be more descriptive and couldn't come up with a good name that would merge this into one.

> * a string might be clearer, something like importance_reweighting(..., action='add') or importance_reweighting(..., action='replace')

I like this option. This would also be easily expandable to other actions.

Reply:
Changed accordingly in [bb7a013](https://github.com/williamjameshandley/anesthetic/pull/122/commits/bb7a013b6ecf386f9e40e96849f30ef9f2e6ad34).

Any further suggestions which other actions might be useful?

Reply:
added `action='mask'` in [45562a0](https://github.com/williamjameshandley/anesthetic/pull/122/commits/45562a086fd397d029c87e7bc467d5f06ffd8d7d).